packages/yew-macro/src/html_tree/html_if.rs

```diff
@@ -0,0 +1,132 @@
+use super::{HtmlRootBraced, ToNodeIterator};
+use crate::PeekValue;
+use boolinator::Boolinator;
+use proc_macro2::TokenStream;
+use quote::{quote_spanned, ToTokens};
+use syn::buffer::Cursor;
+use syn::parse::{Parse, ParseStream};
+use syn::spanned::Spanned;
+use syn::{Expr, Token};
+
+pub struct HtmlIf {
+    if_token: Token![if],
+    cond: Box<Expr>,
+    then_branch: HtmlRootBraced,
+    else_branch: Option<(Token![else], Box<HtmlRootBracedOrIf>)>,
+}
+
+impl PeekValue<()> for HtmlIf {
+    fn peek(cursor: Cursor) -> Option<()> {
+        let (ident, _) = cursor.ident()?;
+        (ident == "if").as_option()
+    }
+}
+
+impl Parse for HtmlIf {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let if_token = input.parse()?;
+        let cond: Box<Expr> = input.parse()?;
+
+        if input.is_empty() {
+            return Err(syn::Error::new(
+                cond.span(),
+                "expected block after this condition",
+            ));
+        }
+
+        let then_branch = input.parse()?;
+        let else_branch = input
+            .parse::<Token![else]>()
+            .ok()
+            .map(|else_token| {
+                if input.is_empty() {
+                    return Err(syn::Error::new(
+                        else_token.span(),
+                        "expected block or `if` token after this token",
```

Comment:
```suggestion
                        "expected block or `if` after `else`",
```
I don't think the error needs to mention tokens and 'this token' will always be 'else' :D 
Before: 
```bash
error: expected block or `if` token after this token
  --> examples/counter/src/main.rs:47:53
   |
47 |         let _x = html! { if true {} else if true {} else };
   |                                                     ^^^^
```
After: 
```bash
error: expected block or `if` after `else`
  --> examples/counter/src/main.rs:47:53
   |
47 |         let _x = html! { if true {} else if true {} else };
   |                                                     ^^^^

```